polkadb/database.go

```diff
@@ -18,76 +18,111 @@ package polkadb
-type BadgerService struct {
+// Db contains directory path to data and db instance
+type Db struct {
 	config Config
 	db     *badger.DB
 }
 
-// ChainDB contains both databases for service registry
-type ChainDB struct {
-	StateDB *BadgerService
-	BlockDB *BadgerService
-	err     <-chan error
-}
+// DbService contains both databases for service registry
+type DbService struct {
+	StateDB *StateDB
+	BlockDB *BlockDB
 
-//Config defines configurations for BadgerService instance
-type Config struct {
-	DataDir string
+	err <-chan error
 }
 
-// Iterable struct contains a transaction, iterator and context fields released, initialized
-type Iterable struct {
-	txn      *badger.Txn
-	iter     *badger.Iterator
-	released bool
-	init     bool
+// BlockDB contains badger.DB instance
+type BlockDB struct {
+	Db *Db
 }
 
-// Batch struct contains a database instance, key-value mapping for batch writes and length of item value for batch write
-type batchWriter struct {
-	db   *BadgerService
-	b    map[string][]byte
-	size int
-}
-
-type table struct {
-	db     Database
-	prefix string
+// StateDB contains badger.DB instance
+type StateDB struct {
+	Db *Db
 }
 
-type tableBatch struct {
-	batch  Batch
-	prefix string
+//Config defines configurations for BadgerService instance
+type Config struct {
+	DataDir string
 }
 
-func (chainDB *ChainDB) Start() <-chan error {
-	chainDB.err = make(<-chan error)
-	return chainDB.err
+// Start starts running BlockDB and StateDB instances
+func (dbService *DbService) Start() <-chan error {
+	dbService.err = make(<-chan error)
+	return dbService.err
 }
 
-func (chainDB *ChainDB) Stop() <-chan error {
+// Stop kills running BlockDB and StateDB instances
+func (dbService *DbService) Stop() <-chan error {
 	e := make(chan error)
-	// Closing Badger Database
-	err := chainDB.StateDB.db.Close()
+	// Closing Badger Databases
+	err := dbService.StateDB.Db.db.Close()
 	if err != nil {
 		e <- err
 	}
-	err = chainDB.BlockDB.db.Close()
+
+	err = dbService.BlockDB.Db.db.Close()
 	if err != nil {
 		e <- err
 	}
 	return e
 }
 
-// NewBadgerService opens and returns a new DB object
-func NewBadgerService(file string) (*BadgerService, error) {
+// NewDatabaseService opens and returns a new DB object
+func NewDatabaseService(file string) (*DbService, error) {
+	stateDataDir := filepath.Join(file, "state")
+	blockDataDir := filepath.Join(file, "block")
+
+	stateDb, err := NewStateDB(stateDataDir)
+	if err != nil {
+		log.Crit("failed to instantiate StateDB", "error", err)
+	}
+
+	blockDb, err := NewBlockDB(blockDataDir)
+	if err != nil {
+		log.Crit("failed to instantiate BlockDB", "error", err)
+	}
+
+	return &DbService{
+		StateDB: stateDb,
+		BlockDB: blockDb,
+	}, nil
+}
+
+// NewBlockDB instantiates BlockDB for storing relevant BlockData
+func NewBlockDB(dataDir string) (*BlockDB, error) {
+	db, err := NewBadgerService(dataDir)
+	if err != nil {
+		log.Crit("error", err)
```

Comment:
iirc this will throw a log15 error as it expects a message followed by an even number of arguments(k/v pairs). Same in `NewStateDB`